lib/charms/sdcore_webui/v0/sdcore_management.py

```diff
@@ -0,0 +1,285 @@
+    ManagementUrlAvailable,
+    SdcoreManagementRequires,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class DummySdcoreManagementRequiresCharm(CharmBase):
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.sdcore_management = SdcoreManagementRequires(self, "sdcore-management")
+        self.framework.observe(
+            self.sdcore_management.on.management_url_available,
+            self._on_management_url_available,
+        )
+
+    def _on_management_url_available(self, event: ManagementUrlAvailable):
+        management_url = event.management_url
+        <do something with the url>
+
+
+if __name__ == "__main__":
+    main(DummySdcoreManagementRequiresCharm)
+```
+
+### Provider charm
+The provider charm is the one providing the address to connect to configuration service.
+
+Example:
+```python
+
+from ops.charm import CharmBase, RelationJoinedEvent
+from ops.main import main
+
+from charms.sdcore_webui.v0.sdcore_management import (
+    SdcoreManagementProvides,
+)
+
+class DummySdcoreManagementProvidesCharm(CharmBase):
+    management_url = "http://1.2.3.4:1234"
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.sdcore_management = SdcoreManagementProvides(
+            self, "sdcore-management"
+        )
+        self.framework.observe(
+            self.on.sdcore_management_relation_joined, self._on_sdcore_management_relation_joined
+        )
+
+    def _on_sdcore_management_relation_joined(self, event: RelationJoinedEvent):
+        if self.unit.is_leader():
+            self.sdcore_management.set_management_url(
+                management_url=self.management_url,
+            )
+
+
+if __name__ == "__main__":
+    main(DummySdcoreManagementProvidesCharm)
+```
+
+"""
+
+import logging
+from typing import Dict, Optional
+
+from interface_tester.schema_base import DataBagSchema  # type: ignore[import]
+from ops.charm import CharmBase, CharmEvents, RelationChangedEvent
+from ops.framework import EventBase, EventSource, Handle, Object
+from ops.model import Relation
+from pydantic import BaseModel, Field, AnyUrl, ValidationError
+
+# The unique Charmhub library identifier, never change it
+LIBID = "e879ede87229469981725ef404973ee5"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+logger = logging.getLogger(__name__)
+"""Schemas definition for the provider and requirer sides of the `sdcore_management` interface.
+It exposes two interfaces.schema_base.DataBagSchema subclasses called:
+- ProviderSchema
+- RequirerSchema
+Examples:
+    ProviderSchema:
+        unit: <empty>
+        app: {
+            "management_url": "http://1.2.3.4:1234",
+        }
+    RequirerSchema:
+        unit: <empty>
+        app:  <empty>
+"""
+
+
+class ProviderAppData(BaseModel):
```

Comment:
The name of the class does not match the interface. Not that important but it would be nice to have them aligned

Reply:
Good point. Done